src/xrCommon/xr_smart_pointers.h

```diff
@@ -1,21 +1,38 @@
 #pragma once
 
+#include "../xrCore/xrMemory.h"
 #include <memory>
+#include <functional>
 
 template <typename T>
-using xr_unique_ptr = std::unique_ptr<T>;
+struct xr_custom_deleter
+{
+    void operator()(T* ptr) const noexcept
+    {
+        xr_delete(ptr);
+    }
+};
+
+template <typename T>
+using xr_unique_ptr = std::unique_ptr<T, xr_custom_deleter<T>>;
 
+// TODO: use xr_custom_deleter here
```

Comment:
Can we implement this TODO?

Reply:
I already did, I just forgot to remove this TODO :)